diff --git a/django_root/django/db/models/base.py b/django_root/django/db/models/base.py
index 27d571c..595849d 100644
--- a/django_root/django/db/models/base.py
+++ b/django_root/django/db/models/base.py
@@ -635,6 +635,23 @@ class Model(object):

     delete.alters_data = True

+    def delete_related(self, using=None):
+        using = using or router.db_for_write(self.__class__, instance=self)
+        connection = connections[using]
+        assert self._get_pk_val() is not None, "%s object can't be deleted because its %s attribute is set to None." % (self._meta.object_name, self._meta.pk.attname)
+
+        # Find all the objects than need to be deleted.
+        seen_objs = CollectedObjects()
+        self._collect_sub_objects(seen_objs)
+
+        # get rid of self so it's not deleted
+        del seen_objs.data[self.__class__][self._get_pk_val()]
+
+        # Actually delete the objects.
+        delete_objects(seen_objs, using)
+
+    delete_related.alters_data = True
+
     def _get_FIELD_display(self, field):
         value = getattr(self, field.attname)
         return force_unicode(dict(field.flatchoices).get(value, value), strings_only=True)
