diff --git a/django_root/django/core/mail/backends/locmem.py b/django_root/django/core/mail/backends/locmem.py
index 642bfc4..df057ae 100644
--- a/django_root/django/core/mail/backends/locmem.py
+++ b/django_root/django/core/mail/backends/locmem.py
@@ -21,4 +21,4 @@ class EmailBackend(BaseEmailBackend):
     def send_messages(self, messages):
         """Redirect messages to the dummy outbox"""
         mail.outbox.extend(messages)
-        return len(messages)
+        return [m.message() for m in messages]
