diff --git a/django_root/django/test/testcases.py b/django_root/django/test/testcases.py
index fc88f5c..78a4676 100644
--- a/django_root/django/test/testcases.py
+++ b/django_root/django/test/testcases.py
@@ -280,6 +280,8 @@ class TransactionTestCase(unittest.TestCase):
         else:
             databases = [DEFAULT_DB_ALIAS]
         for db in databases:
+            cursor = connections[db].cursor()
+            cursor.execute("SET CONSTRAINTS ALL IMMEDIATE")
             call_command('flush', verbosity=0, interactive=False, database=db)
 
             if hasattr(self, 'fixtures'):
